ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -445,9 +446,10 @@ ledgerDbPushMany ::
 ledgerDbPushMany trace cfg aps initDb = (repeatedlyM pushAndTrace) aps initDb
   where
     pushAndTrace ap db = do
-      let pushing = Pushing . toRealPoint $ ap
+      let start   = PushStart . toRealPoint . head $ aps
```

Comment:
Can we avoid these partial functions? I think this one was an unaddressed comment from a previous PR.

Reply:
`head` is partial, but `pushAndTrace` will only be called if `aps` has one element or more

Reply:
I would then do a pattern match and write a case `[] -> error "unreachable because blah"`